streaming/src/main/java/org/apache/spark/streaming/util/WriteAheadLogSegment.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.streaming.util;
+
+/**
+ * This is an interface that represent the information required by any implementation of
+ * a WriteAheadLog to read a written record.
+ */
+@org.apache.spark.annotation.DeveloperApi
+public interface WriteAheadLogSegment extends java.io.Serializable {
```

Comment:
Actually, this is an interface, so I am not sure we can create an alternate
method without breaking binary compatibility.
Well, they could leave the serialization of MyWALSegment to Java, which is
just a wrapper for a ByteBuffer/byte array which contains all the real
data. If that sounds too complicated, then may be we should do bytes. And
probably we should simply use byte array instead of ByteBuffer, as we
probably dont need to deal with direct byte buffers here.


On Mon, Apr 27, 2015 at 9:31 PM, Patrick Wendell <notifications@github.com>
wrote:

> In
> streaming/src/main/java/org/apache/spark/streaming/util/WriteAheadLogSegment.java
> <https://github.com/apache/spark/pull/5645#discussion_r29213656>:
>
> > + *
> > + * Unless required by applicable law or agreed to in writing, software
> > + * distributed under the License is distributed on an "AS IS" BASIS,
> > + * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > + * See the License for the specific language governing permissions and
> > + * limitations under the License.
> > + */
> > +
> > +package org.apache.spark.streaming.util;
> > +
> > +/**
> > + * This is an interface that represent the information required by any implementation of
> > + * a WriteAheadLog to read a written record.
> > + */
> > +@org.apache.spark.annotation.DeveloperApi
> > +public interface WriteAheadLogSegment extends java.io.Serializable {
>
> But in the current approach, they can't for instance use kryo or protobuf
> to serialize, unless they do something really crazy like use an
> externalizable hook to then call into Kryo. I guess I'm just thinking ahead
> to how this will evolve. However, if we want to have this in the future we
> can always create an alternative version that is additive, so I don't feel
> strongly at all
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/5645/files#r29213656>.
>